pkg/config/deployment_test.go

```diff
@@ -379,3 +379,62 @@ func TestTrueByDefaultBoolConfiguration(t *testing.T) {
 		})
 	}
 }
+
+func TestGenericPostSyncConfiguration(t *testing.T) {
+	testcases := []struct {
+		fileName           string
+		expectedKind       Kind
+		expectedAPIVersion string
+		expectedSpec       interface{}
+		expectedError      error
+	}{
+		{
+			fileName:           "testdata/application/generic-postsync.yaml",
+			expectedKind:       KindKubernetesApp,
+			expectedAPIVersion: "pipecd.dev/v1beta1",
+			expectedSpec: &KubernetesDeploymentSpec{
+				GenericDeploymentSpec: GenericDeploymentSpec{
+					Timeout: Duration(6 * time.Hour),
+					Trigger: Trigger{
+						OnOutOfSync: OnOutOfSync{
+							Disabled: newBoolPointer(true),
+						},
+					},
+					PostSync: &PostSync{
+						DeploymentChain: &DeploymentChain{
+							Nodes: []*DeploymentChainNode{
+								{
+									AppName: "app-1",
+								},
+								{
+									AppLabels: map[string]string{
+										"env": "staging",
+										"foo": "bar",
+									},
+								},
+								{
+									AppKind: "ECSApp",
+								},
+							},
+						},
+					},
+				},
+				Input: KubernetesDeploymentInput{
+					AutoRollback: newBoolPointer(true),
+				},
+			},
+			expectedError: nil,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.fileName, func(t *testing.T) {
+			cfg, err := LoadFromYAML(tc.fileName)
```

Comment:
btw, I'm wondering if `config.LoadFromYAML` should accept `fs.FS` as a first argument so that it doesn't get needed to have the actual testdata on disk.

I know managing test data as files is good for readability, but there are many cases where you want to complete in-memory.

Reply:
sure, will change all others place later with your suggestion 👍

Reply:
It's worth considering. Let's make changes actually once got an agreement.